### PR TITLE
Reduced allocations by using &str instead of String

### PR DESCRIPTION
**Pull request by imor**
This PR reduces potential allocations by using string slices (`&str`) instead of owned strings (`String`) in many API methods. Note that this is a breaking change and might need a major version bump.

### CODE REVIEW

File: src/models.rs [hunk 1]
```diff
@@ -150,28 +150,28 @@ pub enum LoginOptions {
...
 
 #[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
-pub struct SignUpWithPhoneAndPasswordPayload {
-    pub(crate) phone: String,
-    pub(crate) password: String,
+pub struct SignUpWithPhoneAndPasswordPayload<'a> {
+    pub(crate) phone: &'a str,
+    pub(crate) password: &'a str,
     pub(crate) options: Option<SignUpWithPasswordOptions>,
 }
```
Comment (imor) [inline src/models.rs:176]: Not part of this PR but consider making these structs private because they are only used internally as payloads for the http calls.

Comment (Proziam) [reply src/models.rs:176]: Thanks you for helping polish things up!

CI doesn't like anything pre-merge at the moment, but this looks fine so I'll merge and fix if anything bubbles up.
